extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java

```diff
@@ -81,4 +103,80 @@ private void addReflectiveHierarchyClass(DotName className) {
     private void addReflectiveClass(boolean methods, boolean fields, String... className) {
         reflectiveClass.produce(new ReflectiveClassBuildItem(methods, fields, className));
     }
+
+    // Generate a ObjectMapperCustomizer bean that registers each serializer / deserializer with ObjectMapper
+    @BuildStep
+    void generateCustomizer(BuildProducer<GeneratedBeanBuildItem> generatedBeans,
+            List<CustomJacksonSerDeserBuildItem> customSerDesers) {
+
+        if (customSerDesers.isEmpty()) {
+            return;
+        }
+
+        ClassOutput classOutput = new ClassOutput() {
+            @Override
+            public void write(String name, byte[] data) {
+                generatedBeans.produce(new GeneratedBeanBuildItem(name, data));
+            }
+        };
+
+        try (ClassCreator classCreator = ClassCreator.builder().classOutput(classOutput)
+                .className("io.quarkus.jackson.customizer.RegisterSerializersAndDeserializersCustomizer")
+                .interfaces(ObjectMapperCustomizer.class.getName())
+                .build()) {
+            classCreator.addAnnotation(Singleton.class);
+
+            try (MethodCreator customize = classCreator.getMethodCreator("customize", void.class, ObjectMapper.class)) {
+                ResultHandle objectMapper = customize.getMethodParam(0);
+
+                for (CustomJacksonSerDeserBuildItem customSerDeser : customSerDesers) {
+                    if (customSerDeser.getItems().isEmpty()) {
```

Comment:
I also wonder if we should use Jandex to discover the ones declared in the application automatically?

What I did in HV is that I detect them automatically if they are beans.

Reply:
That's a very interesting point.

It would work fine for jsonb but probably be tricky for Jackson since as I mentioned earlier the serializers and deserializers need to be part of a module.
We could or course add a one with our own name.

I'm not sure what is best... Having BuildItems or Beans as the starting point...